diesel_codegen/src/schema_inference/pg.rs

```diff
@@ -38,11 +40,11 @@ table! {
 }
 
 pub fn determine_column_type(cx: &mut ExtCtxt, attr: &ColumnInformation) -> P<ast::Ty> {
-    let tpe = if attr.type_name.starts_with("_") {
-        let subtype = str_to_ident(&capitalize(&attr.type_name[1..]));
+    let tpe = if attr.type_name.ends_with("[]") {
```

Comment:
Seems like we could just do a separate query to get the primary key name? (Which is what we ultimately need). This would result in much less code changing to support it.

```rust
fn primary_key_name(conn: &PgConnection, table_oid: u32) -> Vec<String> {
    pg_index
        .inner_join(pg_attribute)
        .select(attname)
        .filter(indisprimary.eq(true))
        .filter(attrelid.eq(table_oid))
        .load(&connection)
}
```